src/main/java/frc/robot/subsystems/Dashboard.java

```diff
@@ -0,0 +1,335 @@
+        });
+
+        initializeSetting("Shooter Bed Offset", Constants.ShooterBed.BED_ANGLE_OFFSET.getDegrees(), _ShooterOffset, value ->
+        {
+            shooterBed.setAngleOffset(Rotation2d.fromDegrees(value));
+        });
+
+        initializeSetting("Max Flywheel Velocity", Constants.ShooterFlywheel.MAX_FLYWHEEL_SPEED, _maxFlywheelVelocity, value ->
+        {
+            ShooterFlywheel.setMaxFlywheelSpeed(value);
+        });
+
+        initializeSetting("Pickup Intake Speed", Constants.Intake.INTAKE_DEFAULT_PERCENT_OUTPUT, _PickupIntakeSpeed, value ->
+        {
+            notepath.setNotepathIntakePickupPercentOutput(value);
+        });
+
+        initializeSetting("Shooter Intake Speed", Constants.Intake.INTAKE_DEFAULT_PERCENT_OUTPUT, _ShooterIntakeSpeed, value ->
+        {
+            notepath.setNotepathShooterPickupPercentOutput(value);
+        });
+
+        initializeSetting("Bed Minimum Angle", Constants.ShooterBed.MIN_BED_ANGLE.getDegrees(), _bedMinimumAngle, value ->
+        {
+            shooterBed.setMinAngle(Rotation2d.fromDegrees(value));
+        });
+
+        initializeSetting("Bed Maximum Angle", Constants.ShooterBed.MAX_BED_ANGLE.getDegrees(), _bedMaximumAngle, value ->
+        {
+            shooterBed.setMaxAngle(Rotation2d.fromDegrees(value));
+        });
+
+        initializeSetting("Max Flywheel Speed", Constants.ShooterFlywheel.MAX_FLYWHEEL_SPEED, _maxFlywheelSpeed, value ->
+        {
+            ShooterFlywheel.setMaxFlywheelSpeed(value);
+        });
+
+        initializeSetting("Flywheel Intake Speed", Constants.Intake.INTAKE_DEFAULT_PERCENT_OUTPUT, _FlywheelIntakeSpeed, value ->
+        {
+            ShooterFlywheel.setFlywheelIntakeSpeed(value);
+        });
+
+    }
+
+    public void initializeSetting(String key, double defaultValue, GenericEntry entry, DoubleConsumer consumer)
+    {
+        NetworkTableInstance.getDefault().addListener(entry, EnumSet.of(NetworkTableEvent.Kind.kValueRemote), event ->
+        {
+            consumer.accept(entry.getDouble(defaultValue));
+            Preferences.setDouble(key, entry.getDouble(defaultValue));
+        });
+
+        double value = defaultValue;
+
+        if (Preferences.containsKey(key))
+        {
+            value = Preferences.getDouble(key, defaultValue);
+        }
+        else
+        {
+            Preferences.initDouble(key, defaultValue);
+        }
+
+        consumer.accept(value);
+        entry.setDouble(value);
+    }
+
+    @Override
+    public void periodic()
+    {
+        var isBlue = false;
+
+        var alliance = DriverStation.getAlliance();
+
+        if (alliance.isPresent() && alliance.get() == DriverStation.Alliance.Blue)
+        {
+            isBlue = true;
+        }
+        else
+        {
+            isBlue = false;
+        }
+
+        SwerveModuleState[] moduleStates = _drive.getModuleStates();
+
+        _allianceBox.setBoolean(isBlue);
```

Comment:
Updated setting of the box:

```suggestion
        _allianceBox.setBoolean(alliance.isPresent() && alliance.get() == DriverStation.Alliance.Blue);
```